app/uk/gov/hmrc/helloworldupscan/repository/UserSessionRepository.scala

```diff
@@ -16,82 +16,61 @@
 
 package uk.gov.hmrc.helloworldupscan.repository
 
-import javax.inject.Inject
+
+import org.bson.types.ObjectId
+import org.mongodb.scala.model.Filters.equal
+import org.mongodb.scala.model.FindOneAndUpdateOptions
+import org.mongodb.scala.model.Updates.set
 import play.api.libs.json._
-import play.modules.reactivemongo.ReactiveMongoComponent
-import reactivemongo.bson.BSONObjectID
-import reactivemongo.play.json.ImplicitBSONHandlers._
 import uk.gov.hmrc.helloworldupscan.connectors.Reference
 import uk.gov.hmrc.helloworldupscan.model._
 import uk.gov.hmrc.helloworldupscan.repository.UploadDetails._
-import uk.gov.hmrc.mongo.ReactiveRepository
-import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.formats.MongoFormats
+import uk.gov.hmrc.mongo.play.json.{Codecs, PlayMongoRepository}
 
+import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-case class UploadDetails(id : BSONObjectID, uploadId : UploadId, reference : Reference, status : UploadStatus)
+case class UploadDetails(_id: ObjectId, uploadId: UploadId, reference: Reference, status: UploadStatus)
```

Comment:
Can we keep the `id` name and just provide an explicit (non-macro) Format? I think the model and the mapping should be decoupled.